python/graphstorm/gconstruct/remap_result.py

```diff
@@ -49,8 +66,62 @@
 # id_maps to each worker process.
 id_maps = {}
 
+def write_data_parquet_file(data, file_prefix, col_name_map=None):
+    """ Write data into disk using parquet format.
+
+        Parameters
+        ----------
+        data: dict of numpy Arrays
+            Data to be written into disk.
+        file_prefix: str
+            File prefix. The output will be <file_prefix>.parquet.
+        col_name_map: dict
+            A mapping from builtin column name to user defined column name.
+    """
+    if col_name_map is not None:
+        data = {col_name_map[key]: val for key, val in data.items()}
+    output_fname = f"{file_prefix}.parquet"
+    write_data_parquet(data, output_fname)
+
+def write_data_csv_file(data, file_prefix, delimiter=",", col_name_map=None):
+    """ Write data into disk using csv format.
+
+        Multiple values for a field are specified with a semicolon (;) between values.
+
+        Example:
+
+        .. code::
+
+            nide, emb
+            0, 0.001;1.2000;0.736;...
+
+        Parameters
+        ----------
+        data: dict of numpy Arrays
+            Data to be written into disk.
+        file_prefix: str
+            File prefix. The output will be <file_prefix>.parquet.
+        delimiter: str
+            Delimiter used to separate columns.
+        col_name_map: dict
+            A mapping from builtin column name to user defined column name.
+    """
+    if col_name_map is not None:
+        data = {col_name_map[key]: val for key, val in data.items()}
+
+    output_fname = f"{file_prefix}.csv"
+    csv_data = {}
+    for key, vals in data.items():
+        if len(vals.shape) == 1:
+            csv_data[key] = vals.tolist()
+        elif len(vals.shape) == 2:
+            csv_data[key] = [";".join([str(v) for v in val]) \
+                             for val in vals.tolist()]
```

Comment:
what is this? `val` is a float?

Reply:
val can be any primitive. If write_data_csv_file is called for saving embeddings, val will be a float.
If write_data_csv_file is called for saving prediction results, it can be integer.